sdk/eventhub/Azure.Messaging.EventHubs/src/Diagnostics/EventHubsEventSource.cs

```diff
@@ -385,23 +385,122 @@ public void GetPropertiesComplete(string eventHubName)
             }
         }
+        /// <summary>
+        ///   Indicates that reading events from an Event Hub partition has started.
+        /// </summary>
+        ///
+        /// <param name="eventHubName">The name of the Event Hub that events are being read from.</param>
+        /// <param name="partitionId">The identifier of the partition that properties are being read from.</param>
+        ///
+        [Event(21, Level = EventLevel.Informational, Message = "Beginning to read events for Event Hub: {0} (Partition Id: '{1}').")]
+        public void ReadEventsFromPartitionStart(string eventHubName,
+                                                 string partitionId)
+        {
+            if (IsEnabled())
+            {
+                WriteEvent(21, eventHubName ?? string.Empty, partitionId ?? string.Empty);
+            }
+        }
+
+        /// <summary>
+        ///   Indicates that reading events from an Event Hub partition has completed.
+        /// </summary>
+        ///
+        /// <param name="eventHubName">The name of the Event Hub that events are being read from.</param>
+        /// <param name="partitionId">The identifier of the partition that properties are being read from.</param>
+        ///
+        [Event(22, Level = EventLevel.Informational, Message = "Completed reading events for Event Hub: {0} (Partition Id: '{1}').")]
+        public void ReadEventsFromPartitionComplete(string eventHubName,
+                                                    string partitionId)
+        {
+            if (IsEnabled())
+            {
+                WriteEvent(22, eventHubName ?? string.Empty, partitionId ?? string.Empty);
+            }
+        }
+
+        /// <summary>
+        ///   Indicates that an exception was encountered while reading events from an Event Hub partition.
+        /// </summary>
+        ///
+        /// <param name="eventHubName">The name of the Event Hub that events are being read from.</param>
+        /// <param name="partitionId">The identifier of the partition that properties are being read from.</param>
+        /// <param name="errorMessage">The message for the exception that occurred.</param>
+        ///
+        [Event(23, Level = EventLevel.Error, Message = "An exception occurred while reading events for Event Hub: {0} (Partition Id: '{1}'). Error Message: '{2}'")]
+        public void ReadEventsFromPartitionError(string eventHubName,
+                                                 string partitionId,
+                                                 string errorMessage)
+        {
+            if (IsEnabled())
+            {
+                WriteEvent(23, eventHubName ?? string.Empty, partitionId ?? string.Empty, errorMessage ?? string.Empty);
+            }
+        }
+
+        /// <summary>
+        ///   Indicates that reading events from an Event Hub partition has started.
```

Comment:
It should be `from all partitions of an Event Hub`.  We need the same change for Complete and Error methods.